cmd/zc_traverser_blob.go

```diff
@@ -95,6 +97,26 @@ func (t *blobTraverser) getPropertiesIfSingleBlob() (props *azblob.BlobGetProper
 	return nil, false, false, err
 }
 
+func (t *blobTraverser) getBlobTags(requestID *string, ifTags *string) (common.BlobTags, error) {
+	// trim away the trailing slash before we check whether it's a single blob
+	// so that we can detect the directory stub in case there is one
+	blobUrlParts := azblob.NewBlobURLParts(*t.rawURL)
+	blobUrlParts.BlobName = strings.TrimSuffix(blobUrlParts.BlobName, common.AZCOPY_PATH_SEPARATOR_STRING)
+
+	// perform the check
+	blobURL := azblob.NewBlobURL(blobUrlParts.URL(), t.p)
+	blobTagsMap := make(common.BlobTags)
+	blobGetTagsResp, err := blobURL.GetTags(t.ctx, nil, requestID, nil, nil, ifTags)
```

Comment:
>blobGetTagsResp, err := blobURL.GetTags(t.ctx, nil, requestID, nil, nil, ifTags) [](start = 1, length = 80)

Oh wait, GetTags is taking in timeout *int32, requestID *string, snapshot *string, versionID *string, this doesn't look right, since the timeout/requestID are not supposed to be exposed, and snapshot/versionID are on the URL.

Reply:
This should be fixed in the SDK.

---
In reply to: [569209305](https://github.com/Azure/azure-storage-azcopy/pull/1280#discussion_r569209305) [](ancestors = 569209305)

Reply:
Yeah the enlightenment about not exposing RequestID/Timeout came to me pretty recently. I'll fix this in SDK.

Reply:
https://github.com/Azure/azure-storage-blob-go/pull/252 - Here is the fix for Blob SDK